hyperspace/parachain/src/light_client_sync.rs

```diff
@@ -275,9 +275,6 @@ where
 		})
 		.collect::<BTreeSet<_>>();
 
-	// We always insert the latest finalized para height
-	headers_with_events.insert(T::BlockNumber::from(latest_finalized_para_height));
```

Comment:
why removing this?

Reply:
I added a check below to ensure the we only add this when the `previous finalized parachain` height is less than the `latest_finalized_para_height`, if not it could lead to some errors when verifying the client update